spec/unit/lib/wifi_user/use_cases/sponsor_users_spec.rb

```diff
@@ -8,10 +8,18 @@
   let(:staging_do_not_reply_id) { '45d6b6c4-6a36-47df-b34d-256b8c0d1511' }
 
   let(:user_model) { double(generate: { username: username, password: password }) }
-  subject { described_class.new(user_model: user_model) }
+  let(:whitelist_checker) { double(execute: { success: true }) }
+
+  subject do
+    described_class.new(
+      user_model: user_model,
+      whitelist_checker: whitelist_checker
+    )
+  end
 
   before do
     ENV['RACK_ENV'] = environment
+    ENV['NOTIFY_API_KEY'] = 'dummy_key-00000000-0000-0000-0000-000000000000-00000000-0000-0000-0000-000000000000'
```

Comment:
Noticed this test was failing when run on its own.
Depended on setup from a previous test to set this env variable.

Needs looking into, will do in a separate pr

Reply:
Agreed.
we should do something similar for Env with as we did with the s3 config.
This can act as a reference so we can enforce each example is clean: https://www.rubydoc.info/github/rspec/rspec-support/RSpec%2FSupport%2FShellOut:with_env

Reply:
Haven't seen that, looks cool - can you link to where else we took that approach @cybojenix ?

Reply:
We've done it with a closed source solution in the past, but it would amount to something along the lines of:

```rb
  let(:env_override) do
    { NOTIFY_API_KEY: '' }
  end
  around do |example|
    with_env(env_override) { example.call }
  end
```